discovery/discovery.go

```diff
@@ -93,12 +99,18 @@ func (o *orchestratorPool) GetOrchestrators(numOrchestrators int) ([]*net.Orches
 
 	select {
 	case <-ctx.Done():
+		respLock.Lock()
+		returnOrchs := orchInfos[:numOrchestrators]
+		respLock.Unlock()
 		glog.Info("Done fetching orch info for orchestrators, context timeout: ", orchInfos)
 		cancel()
-		return orchInfos, nil
+		return returnOrchs, nil
 	case <-orchChan:
+		respLock.Lock()
+		returnOrchs := orchInfos[:numOrchestrators]
+		respLock.Unlock()
 		glog.Info("Done fetching orch info for orchestrators, numResponses fetched: ", orchInfos)
```

Comment:
There's still the possibility of `orchInfos` mutation here so the log should print `returnOrchs`